UnityProject/Assets/Scripts/Mapping/EventRouter.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using SecureStuff;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class EventRouter : MonoBehaviour
+{
+	//TODO Specifying data sometime
+	//why Singular, gets complicated if you add multiple since you have to do a Equality of the elements of the list
+	public EventConnection Connection1 = new EventConnection();
+	public EventConnection Connection2 = new EventConnection();
+	public EventConnection Connection3 = new EventConnection();
+	public EventConnection Connection4 = new EventConnection();
+	public EventConnection Connection5 = new EventConnection();
```

Comment:
Why not use a list

Reply:
the map saver, Would have to be expanded to support class comparison ( is the class exactly the same or is it different ) 
and then be able to serialise the differences if it's only slightly different, basically big pain in but

Reply:
so, with the list, 
it needs to be able to compare with the prefab yes? , how do you compare a class?

Reply:
because all it has is the object and type, I was thinking may be supporting structs but that seems like a dodgy implementation

Reply:
you'd have to write a custom comparer for each class as well  since its something not standard with classes